subsys/net/ip/net_context.c

```diff
@@ -689,6 +689,45 @@ int net_context_put(struct net_context *context)
 	return 0;
 }
 
+static int net_context_bind_default(struct net_context *context,
+				    const struct sockaddr *dest_addr)
```

Comment:
Having dest_addr here is great, but the function is not using it for anything currently. So the code is missing a call to 
```
src_addr = net_if_ipv6_select_src_addr(NULL, dest_addr);
```
and then the src_addr can be used to get the interface by calling
```
if (!net_if_ipv6_addr_lookup(src_addr, &iface)) {
        return -EINVAL;
}
```
If the caller does not supply the dest_addr, then we can use the default interface. For IPv4 there is no source address selection function currently, so we can just use the default interface for IPv4.